content/source/docs/enterprise-beta/registry/publish.html.md

```diff
@@ -0,0 +1,96 @@
+your configured [VCS providers][vcs], and TFE's VCS user account must have admin
+access to the repository. (Since the registry uses webhooks to import new module
+versions, it needs admin access to create those webhooks.)
+
+- **Named `terraform-<PROVIDER>-<NAME>`.** Module repositories must use this
+three-part name format, where `<NAME>` reflects the type of infrastructure the
+module manages and `<PROVIDER>` is the main provider where it creates that
+infrastructure. The `<NAME>` segment can contain additional hyphens. Examples:
+`terraform-google-vault` or `terraform-aws-ec2-instance`.
+
+- **Standard module structure.** The module must adhere to the
+[standard module structure](/docs/modules/create.html#standard-module-structure).
+This allows the registry to inspect your module and generate documentation,
+track resource usage, and more.
+
+- **`x.y.z` tags for releases.** The registry uses tags to identify module
+versions. Release tag names must be a [semantic version](http://semver.org),
+which can optionally be prefixed with a `v`. For example, `v1.0.4` and `0.9.2`.
+To publish a module initially, at least one release tag must be present. Tags
+that don't look like version numbers are ignored.
+
+## Publishing a New Module
+
+To publish a module, navigate to the modules list with the "Modules" button and click the "+ Add Module" button in the upper right.
+
+![TFE screenshot: the "modules" button and the "+Add Module" button](./images/publish-add-button.png)
+
+This brings you to the "Add a New Module" page, which has a text field and at least one VCS provider button.
+
+![TFE screenshot: the "add a new module" page, with a repository name entered](./images/publish-add-module.png)
+
+If you have multiple VCS providers configured, use the buttons to select one. In the text field, enter the name of the repository for the module you're adding. Then click the "Publish Module" button.
+
+~> **Note:** The name you type into the repo field will usually be something like `hashicorp/terraform-aws-vpc` or `INFRA/terraform-azure-appserver`. Module repo names use a `terraform-<PROVIDER>-<NAME>` format, and VCS providers use `<NAMESPACE>/<REPO NAME>` strings to locate repositories. (For most providers the namespace is a username or organization name, but Bitbucket Server uses project keys, like `INFRA`.)
+
+TFE will display a loading page while it imports the module versions from version control, and will then take you to the new module's details page. On the details page you can view available versions, read documentation, and copy a usage example.
+
+![TFE screenshot: the module loading page](./images/publish-loading.png)
+
+![TFE screenshot: a module details page](./images/publish-module-details.png)
+
+## Releasing New Versions of a Module
+
+To release a new version of a module, push a new version tag to its VCS repository. The registry will automatically import the new version.
```

Comment:
It could be helpful to mention that this can be done either from the VCS provider's UI (like GitHub releases), or by adding/pushing tags from the git CLI. It's common to get hung up on the CLI while pushing tags, because you have to run `git push --tags` in order for the tags to get pushed up properly.